### PR TITLE
feat!: require chain id in transactions

### PR DESCRIPTION
**Pull request by joroshiba**
## Summary
Adds the concept of `TransactionParams` to `UnsignedTransaction` as a body that can grow to include many different parameters which much pass for a transaction to be valid, and uses this structure to add a `chain_id` to ensure that transactions from one Astria network cannot be replayed on another.

## Background
There are likely to be additional parameters we can add onto transactions (such a expiration or things relating to auction bids), and we needed to add the chain id parameter.

## Changes
- Modify Transaction format
- Add chain id param and checks.

## Testing
CI/CD

### CODE REVIEW

**Review by aajimal — APPROVED** [review]

File: crates/astria-composer/local.env.example
Comment (SuperFluffy) [inline crates/astria-composer/local.env.example:31]: IMO remove in favor of querying sequencer's `/genesis` endpoint to extract the cometbft chain ID from there. This is how conductor is doing it and I would like this to be consistent between services.

File: crates/astria-sequencer/src/transaction/mod.rs
Comment (SuperFluffy) [inline crates/astria-sequencer/src/transaction/mod.rs:177]: why `pub(crate)`?

**Review by SuperFluffy — CHANGES_REQUESTED** [review]
The chain ID is already available when connecting to sequencer by querying it's ABCI `/genesis` endpoint, which is what conductor is doing.

I would like our services to be consistent in how they query the chain ID and use a single source of truth (the mentioned endpoint).

However, if you prefer to explicitly use a configuration parameter for this (for example to guard against accidentally connecting to the wrong sequencer network), then please still query `/genesis` and check that `configured_chain_id == remote_chain_id` to have composer *before* it receives or polls any transactions.

File: crates/astria-core/src/protocol/abci.rs
Comment (SuperFluffy) [inline crates/astria-core/src/protocol/abci.rs:19]: Please add an entry to `AbciErrorCode::info` below

File: crates/astria-core/src/protocol/abci.rs
Comment (SuperFluffy) [inline crates/astria-core/src/protocol/abci.rs:1]: This should also get a matching entry in the `AbciErrorCode` protobuf enum

**Review by SuperFluffy — APPROVED** [review]
Approving after syncing offline. We decided to a) require Sequencer chain IDs be set in config, and b) the configured Sequencer chain IDs be checked against the `/genesis` endpoint. However, this will and can be done in follow PRs (for both composer and conductor).

I am approving this PR. Before merge, please address my comments re the ABCI error codes:

1. add entries to the `Display` and `From` impls
2. add matching protobuf ABCI error codes (note: apparently these have been removed https://github.com/astriaorg/astria/pull/869 for some reason)

File: crates/astria-sequencer/src/transaction/mod.rs
Comment (joroshiba) [reply crates/astria-sequencer/src/transaction/mod.rs:177]: Admittedly I just matched the InvalidNonce error publicity since these should be consumed similarly.

File: crates/astria-core/src/protocol/abci.rs
Comment (joroshiba) [reply crates/astria-core/src/protocol/abci.rs:1]: There is no longer a matching protobuf enum

**Comment by joroshiba** [comment]
Follow up issue: https://github.com/astriaorg/astria/issues/986

File: crates/astria-sequencer/src/transaction/mod.rs
Comment (noot) [inline crates/astria-sequencer/src/transaction/mod.rs:281]: you can remove this - the mempool check is sufficient, no txs with invalid chain ids will make it here. the nonce check below is needed for strict equality before execution

File: crates/astria-sequencer/src/app.rs
Comment (noot) [inline crates/astria-sequencer/src/app.rs:2186]: move/change this test to test the mempool check (after you delete the check_stateful check this test will fail)

File: crates/astria-sequencer/src/transaction/mod.rs
Comment (joroshiba) [reply crates/astria-sequencer/src/transaction/mod.rs:281]: I think we need to protect against a validator running their node with a no op mempool. IE someone can modify code for their validator grab alternate transactions and propose?

**Pull request merged**
